fs/fs_zenfs.cc

```diff
@@ -788,6 +788,23 @@ Status ZenFS::DecodeSnapshotFrom(Slice* input) {
   return Status::OK();
 }
 
+/* This function only read through slices in a snapshot record and
+ * DO NOT add valid slices to files_ to avoid duplicating files
+ * appeared in different snapshots*/
+Status ZenFS::DecodeSnapshotFromAndNoCacheFiles(Slice* input) {
```

Comment:
The function signature looks not straightforward to me, and could you please explain a little bit more about why we need this?